ext/opcache/zend_persist_calc.c

```diff
@@ -91,6 +91,14 @@ static void zend_persist_ast_calc(zend_ast *ast)
 		zval z;
 		ZVAL_PTR(&z, zend_ast_get_op_array(ast)->op_array);
 		zend_persist_op_array_calc(&z);
+
+		/* If op_array is shared, the function name refcount is still incremented for each use,
+		 * so we need to release it here. We remembered the original function name in xlat. */
+		zend_string *old_function_name =
+			zend_shared_alloc_get_xlat_entry(&zend_ast_get_op_array(ast)->op_array->function_name);
+		if (old_function_name) {
+			zend_string_release_ex(old_function_name, 0);
+		}
```

Comment:
I didn't understand this why you release something in ``zend_persist_ast_calc()``.
Will this work if after ``zend_persist_calc()`` we will decide not persist the script (e.g. because of OOM)?

Reply:
The name is released in calc because it is also interned in calc. We cannot release it in persist because then the original reference is gone.